jupinx/theme/minimal/templates/html.tpl

```diff
@@ -26,7 +26,9 @@
 		<meta name="keywords" content="">
 		<meta name="description" content="">
 
-		<link rel="stylesheet" href="/_static/css/base.css">
+		<link rel="stylesheet" href="_static/css/base.css">
```

Comment:
thanks @GavinHuttley -- have you by chance tested this in an environment with nested folders? That is the main reason we used `root` referenced stylesheets as a default as one of our projects has <topic>/<lecture.rst> structure.

Reply:
I understand and agree the patch does not solve the nested folder case too. However, neither does the existing code.

I made a small edit to the `demo_notebook.rst` to the content below. Running `$ jupinx -w` produced html without syntax highlighting.

```rst
To add a code block you can use ``code-block`` directives such as:

.. code:: python3

    print("hello")
```
My patch did solve the syntax highlighting for this case. However, when I add a nested `rst`, `source/lectures/lecture01.rst` as per your structure (hooking it up via the `index.rst`), that file does not get styled correctly with either `href="_static...` or `href="/_static...` in the template.

It seems it should be straightforward to specify a relative path to the configured setting in the template (presumably `themes/minimal`, or perhaps `html_static_path`?). However, I'm not familiar enough with building templates to solve this quickly. Some googling suggested the following:
```jinja
<link rel="stylesheet" href="{{ pathto('_static/base.css', 1) }}" type="text/css" />
```
but `pathto` is distributed as part of sphinx and the minimal template is inheriting from a `nbconvert` template. So that's as far as I've gotten....

Reply:
thanks @GavinHuttley -- this is useful. I will update our demo notebooko as I just noticed it is using the old `.. code::` directive rather than ``.. code-block::`` so the extension is probably not picking it up as a code block. 

I will have a deeper think (hopefully today) on the path issues.

Reply:
Actually, that `code` directive usage is mine, not yours -- oops!

Reply:
@GavinHuttley - did the `code-block` change bring in highlighting for you?

Reply:
Unfortunately, no. [Linking to a tarball of the demo project](https://www.dropbox.com/s/v9a2rkvoarfxf45/jupinx-project.tar.gz?dl=0) so we're working off the same material.

Reply:
thanks @GavinHuttley -- that is helpful. What targets are you looking to build? `jupyter` notebooks, `html`, and or `pdf`? Currently we have two builders `jupyter` and `jupyterpdf`. So if you would like to generate the `html` you will need to enable `jupyter_make_site = True` in `conf.py` and allow `notebook_execution`. It currently takes some configuration to get the `website` building from the base quickstart. It is setup to build notebooks by default without execution and website building. This will change in November as we are planning a big refactor: 

https://github.com/QuantEcon/sphinxcontrib-jupyter/wiki/Extension-Refactor-and-Redesign-(November-2019)

is the planning document.

Reply:
@GavinHuttley we actually added configuration overrides in the `Makefile` to allow `jupyter` and `website` building. But I suspect you have been using `make html`? which would use the sphinx html writer.

Reply:
`make website` using the `minimal` theme bundled with `jupinx-quickstart` should give you the following for `lecture` with the `root reference` in place.

![image](https://user-images.githubusercontent.com/8263752/67355667-8caaf800-f5a3-11e9-8362-78654f0a2d67.png)

Reply:
I will update https://jupinx.readthedocs.io/en/latest/quickstart.html with additional details now.

Reply:
https://github.com/QuantEcon/jupinx/commit/c611c620335515fd6d92efb876d0c18231e3ccd1

Reply:
Let me know if that solves you problem: https://jupinx.readthedocs.io/en/latest/quickstart.html

Reply:
I was building the website using `jupinx -w`. After removing `_build`, tried again via `make website`. No change.

For reference, I'm running MacOS 10.14.6 and using conda 4.7.12

In an effort to standardise, I've created a [minimal conda environment file](https://www.dropbox.com/s/pkqykv6c2d2aprc/jupinx_env.yml?dl=0). (As an aside, GitHub is not allowing me to attach this to the issue ... grrr, thus the dropbox link.) I'm starting from an empty directory (having copied `source`).

I'm then doing the following:
```
$ conda env create -f jupinx_env.yml
## after environment created
$ conda activate jupinx
(jupinx) $ python -m ipykernel install --user --name python3
(jupinx) $ rm -rf jupinx-project; mkdir jupinx-project; cd jupinx-project
(jupinx) $ jupinx-quickstart # follow prompts
(jupinx) $ rm -rf source
(jupinx) $ cp -r ../source source
```
I then edit `conf.py` `jupyter_kernels` so it only has `python3`, set `jupyter_execute_notebooks = True` and `jupyter_make_site = True`. Then
```
(jupinx) $ make website
(jupinx) $ make website
Running Sphinx v2.2.0
making output directory... done
WARNING: html_static_path entry 'static' does not exist
building [mo]: targets for 0 po files that are out of date
building [jupyter]: targets for 3 source files that are out of date
updating environment: [new config] 3 added, 0 changed, 0 removed
reading sources... [100%] lectures/lecture01                                                                              
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] lectures/lecture01                                                                               
copying static files... copying static files to executed folder... 
done
Starting notebook execution for website and html conversion(if set in config)...
(1/3)  demo_notebook -- pass -- 1.49s
(2/3)  lecture01 -- pass -- 0.91s
(3/3)  index -- pass -- 0.92s
WARNING: html folder not present in the themes directory
WARNING: coverage directory not found. Please ensure to run coverage build before running website build
build succeeded, 3 warnings.
(jupinx) $ open _build/website/jupyter_html/index.html
```
and checked the output. The `print` statements were executed, but there is no syntax highlighting in either the demo or lecture01 files..

Reply:
thanks for persisting with this @GavinHuttley. You will also need to add

```python
#template names
jupyter_html_template = "html.tpl"
jupyter_latex_template = "latex.tpl"
```

to the `conf.py` file to connect the templates. This is one of the changes we are making in November for more robus theme support. Currently you have to set 4 variables to get everything connected.

Reply:
```
# --------------------------------------------
# jupyter Sphinx Extension conversion settings
# --------------------------------------------

# Conversion Mode Settings
# If "all", convert codes and texts into notebook
# If "code", convert codes only
jupyter_conversion_mode = "all"

jupyter_write_metadata = False

# Location for _static folder
jupyter_static_file_path = ["source/_static"]

# Configure jupyter headers
jupyter_headers = {
    "python3": [
        # nbformat.v4.new_code_cell("%autosave 0")      #@mmcky please make this an option
        ],
    "julia": [
        ],
}

# Filename for the file containing the welcome block
jupyter_welcome_block = ""

#Adjust links to target html (rather than ipynb)
jupyter_target_html = False

#path to download notebooks from 
jupyter_download_nb_urlpath = None

#allow downloading of notebooks
jupyter_download_nb = False

#Use urlprefix images
jupyter_images_urlpath = None

#Allow ipython as a language synonym for blocks to be ipython highlighted
jupyter_lang_synonyms = ["ipython"]

#Execute skip-test code blocks for rendering of website (this will need to be ignored in coverage testing)
jupyter_ignore_skip_test = True

#allow execution of notebooks
jupyter_execute_notebooks = True

# Location of template folder for coverage reports
jupyter_template_coverage_file_path = False

# generate html from IPYNB files
jupyter_generate_html = True

# theme path
jupyter_theme_path = "theme/minimal"

# template path
jupyter_template_path = "theme/minimal/templates"

#template names
jupyter_html_template = "html.tpl"
jupyter_latex_template = "latex.tpl"

#make website
jupyter_make_site = True

#force markdown image inclusion
jupyter_images_markdown = False

#This is set true by default to pass html to the notebooks
jupyter_allow_html_only=True
```

Reply:
I will setup a new environment and see if I can replicate the above.

Reply:
I followed the setup above exacly and I see:

![image](https://user-images.githubusercontent.com/8263752/67440041-bc114180-f643-11e9-85ed-59ea4ddfd1e8.png)

once it is built. Now I am confused. BTW - the `conf.py` is setup to server `jupyter` as the default case and `website` and `pdf` are setup in the Makefile to handle all the setting overrides to include `execution` and `html` generation so you shouldn't have to modify the `conf.py` from the `quickstart` (with the exeception of answering 'n' to language kernels which is fixed in https://github.com/QuantEcon/jupinx/pulls?q=is%3Apr+is%3Aclosed.)

Reply:
if you run the above workflow -- would you mind zipping up the folder again and I will run it here and try to figure this out.

Reply:
Still no success. I restarted from scratch, using the minimal conda yml from yesterday, recreated the `jupinx-project` and ran `make website`. [Here's the requested zip](https://www.dropbox.com/s/ppk007n6i8jlwl8/jupinx-project.zip?dl=0) (GitHub still not letting me attach files.)

I have now also replicated this issue on an Ubuntu (16.04.6) VM I've just set up. This makes me wonder whether you have any settings in your user environment, e.g. `~/.jupyter`, that may be affecting this?

Reply:
thanks @GavinHuttley -- I will use your zip on a new machine to check out if I can replicate this. Thanks again for reporting this.

Reply:
Hi @GavinHuttley. OK I think I understand what is happening. You are opening the file:

```bash
open _build/website/jupyter_html/index.html
```

directly -- however the site should be served using a `http` server to get the `stylesheets` and `js` to load. You can use `jupinx -s` to view the html results which will start a python `http.server` in the `_build/website/jupyter_html` directory for you.

Reply:
otherwise you can use `python -m http.server` in the `_build/website/jupyter_html` folder. Please let me know if that fixes the issue on your end.

Reply:
Ha! That's it, the syntax highlighting works now. I guess I've been thinking about `sphinxcontrib-jupyter` and `jupinx` in terms of generating static documentation for my various projects on readthedocs.